tests/api.c

```diff
@@ -10119,15 +10274,95 @@ static void test_wolfSSL_DES_ecb_encrypt(void)
     AssertIntEQ(ret1,0);
     wolfSSL_DES_ecb_encrypt(&output2,&back2,&key,DES_DECRYPT);
-    ret2 = memcmp((unsigned char *) back2,(unsigned char *) input2,sizeof(WOLFSSL_DES_cblock));
+    ret2 = XMEMCMP((unsigned char *) back2,(unsigned char *) input2,sizeof(WOLFSSL_DES_cblock));
     AssertIntEQ(ret2,0);
 
     printf(resultFmt, passed);
     #endif
 }
+
+static void test_wolfSSL_ASN1_TIME_adj(void)
+{
+#if defined(OPENSSL_EXTRA) && !defined(NO_ASN1_TIME) \
+&& !defined(USER_TIME) && !defined(TIME_OVERRIDES)
+
+    const int year = 365*24*60*60;
+    const int day = 24*60*60;
+    const int hour = 60*60;
+    const int min = 60;
+    const byte asn_utc_time = 0x17;
+    const byte asn_gen_time = 0x18;
+    WOLFSSL_ASN1_TIME *asn_time, *s;
+    int offset_day;
+    long offset_sec;
+    char date_str[20];
+
+    printf(testingFmt, "wolfSSL_ASN1_TIME_adj()");
+
+    s = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL,
+                                    DYNAMIC_TYPE_OPENSSL);
+    /* UTC notation test */
+    /* 2000/2/15 20:30:00 */
+    time_t t = (time_t)30 * year + 45 * day + 20 * hour + 30 * min + 7 * day;
+    offset_day = 7;
+    offset_sec = 45 * min;
+    /* offset_sec = -45 * min;*/
+    asn_time = wolfSSL_ASN1_TIME_adj(s, t, offset_day, offset_sec);
+    AssertTrue(asn_time->data[0] == asn_utc_time);
+    XSTRNCPY(date_str,(const char*) &asn_time->data+2,13);
+    AssertIntEQ(0, XMEMCMP(date_str, "000222211500Z", 13));
+
+    /* negative offset */
+    offset_sec = -45 * min;
+    asn_time = wolfSSL_ASN1_TIME_adj(s, t, offset_day, offset_sec);
+    AssertTrue(asn_time->data[0] == asn_utc_time);
+    XSTRNCPY(date_str,(const char*) &asn_time->data+2,13);
+    AssertIntEQ(0, XMEMCMP(date_str, "000222194500Z", 13));
+
+    XFREE(s,NULL,DYNAMIC_TYPE_OPENSSL);
+    XMEMSET(date_str, 0, sizeof(date_str));
+
+    s = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL,
+                                    DYNAMIC_TYPE_OPENSSL);
+    /* GeneralizedTime notation test */
+    /* 2055/03/01 09:00:00 */ 
+    t = (time_t)85 * year + 59 * day + 9 * hour + 21 * day;
+    offset_day = 12;
+    offset_sec = 10 * min;
+    asn_time = wolfSSL_ASN1_TIME_adj(s, t, offset_day, offset_sec);
+    AssertTrue(asn_time->data[0] == asn_gen_time);
+    XSTRNCPY(date_str,(const char*) &asn_time->data+2, 15);
+    AssertIntEQ(0, XMEMCMP(date_str, "20550313091000Z", 15));
+
+    XFREE(s,NULL,DYNAMIC_TYPE_OPENSSL);
+    XMEMSET(date_str, 0, sizeof(date_str));
+
+    /* if WOLFSSL_ASN1_TIME struct is not allocated */
+    s = NULL;
+
+    t = (time_t)30 * year + 45 * day + 20 * hour + 30 * min + 15 + 7 * day;
+    offset_day = 7;
+    offset_sec = 45 * min;
+    asn_time = wolfSSL_ASN1_TIME_adj(s, t, offset_day, offset_sec);
+    AssertTrue(asn_time->data[0] == asn_utc_time);
+    XSTRNCPY(date_str,(const char*) &asn_time->data+2,13);
+    AssertIntEQ(0, XMEMCMP(date_str, "000222211515Z", 13));
+    XFREE(s,NULL,DYNAMIC_TYPE_OPENSSL);
```

Comment:
Please remove this line. The `s` variable will always be NULL here.

Reply:
I misunderstood your suggestion.
I deleted it.